test/integration/install/install_test.go

```diff
@@ -445,22 +444,25 @@ func TestInstallOrUpgradeCli(t *testing.T) {
 	}
 
 	// Install Linkerd Viz Extension
-	exec = append(vizCmd, vizArgs...)
-	out, err = TestHelper.LinkerdRun(exec...)
-	if err != nil {
-		testutil.AnnotatedFatal(t, "'linkerd viz install' command failed", err)
-	}
+	if TestHelper.Viz() || TestHelper.UpgradeFromVersion() != "" {
```

Comment:
Is it possible to completely remove viz from install_test?  If we have install_test install just the core control plane, then the viz suite can be responsible for installing the viz extension.  Similarly, the external suite can be responsible for installing the viz extension with an external prometheus.  This potentially lets us slim down the massive install_test.go and remove some global state for Viz() and ExternalPrometheus() from TestHelper.

Ultimately, having these test suites be separate and composable would be better than having them coupled and having their behavior controller by flags.

Reply:
Yup! It would definitely be possible. I had planned to do this in a subsequent change. My ideal state to be in at the end is for every suite to have its own install file. This would put us in a position where we no longer need to use a test runner script to run the tests locally or in CI.

I thought about doing it as part of this change, but I think that'd make this PR harder to review. I opted for less changes; the `--viz` flag is a temporary solution to make sure this change is non-obstructive. If you think it'd be better to do it now, I'm also ok with making the changes!

Reply:
I'll leave it up to you how you want to break up and sequence the PRs, but adding a viz flag to the install_test feels like a step backwards rather than an incremental improvement.  My instinct would be to split out viz as early as possible.